### ISSUE TITLE
Mediapipe-model-maker

### ISSUE THREAD

**Issue by sustinespacne9-ai** [google-ai-edge/mediapipe#6244]
> Hi everyone,
> 
> I’m trying to train a custom object detector using tflite-model-maker in Google Colab, but the training fails when it attempts to download the pretrained MobileNetV2 SSD checkpoint.
> 
> The URL being accessed is:
> https://storage.googleapis.com/tf_model_garden/vision/qat/mobilenetv2_ssd_coco/mobilenetv2_ssd_i256_ckpt.tar.gz
> and it returns:
> AccessDenied (storage.objects.get permission denied)
> 
> This happens when I run the training step below:
> 
> hparams = object_detector.HParams(
>     batch_size=4,
>     learning_rate=0.3,
>     epochs=100,
>     export_dir='exported_model'
> )
> 
> options = object_detector.ObjectDetectorOptions(
>     supported_model=object_detector.SupportedModels.MOBILENET_V2,
>     hparams=hparams
> )
> 
> model = object_detector.ObjectDetector.create(
>     train_data=train_data,
>     validation_data=val_data,
>     options=options
> )
> 
> What I’ve already tried:
> 
> Restarting the Colab runtime
> 
> Reinstalling tflite-model-maker (including the nightly version)
> 
> Clearing ~/.keras/models and cache
> 
> Confirmed that the runtime has a stable internet connection
> 
> My questions are:
> 
> Is this pretrained checkpoint no longer publicly accessible?
> 
> Is there an updated download source or a patched version of Model Maker that uses a new path?
> 
> Is tflite-model-maker still supported for object detection, or is the recommended workflow now MediaPipe Model Maker?
> 
> Any clarification or updated example notebook would be greatly appreciated. Thanks! 

 _Originally posted by @sustinespacne9-ai in [#5214](https://github.com/google-ai-edge/mediapipe/issues/5214#issuecomment-3940681210)_

**Comment by piekniewski** [comment]
I have the same issue, did google accidentally delete those models?

**Comment by akashvverma1995** [comment]
Hi @sustinespacne9-ai @piekniewski 

Thank you for reporting the issue. We're looking into it on our side and will keep you informed with any updates.

**Comment by sustinespacne9-ai** [comment]
> I have the same issue, did google accidentally delete those models?

I’m not entirely sure, but it seems possible that the original pretrained weight file was either removed, moved, or its access permissions were changed on the hosting server. Since the error is HTTP 403 (Forbidden), it suggests that the file may still exist but is no longer publicly accessible from the original link.

**Comment by leonardosarti-nando** [comment]
Same issue here

**Comment by karthikeyanelumina** [comment]
@akashvverma1995 - Do we have any updates from the Google team? We’ve been waiting for a resolution on this issue for the past two weeks. Thanks in advance

**Comment by sustinespacne9-ai** [comment]
> [@akashvverma1995](https://github.com/akashvverma1995) - Do we have any updates from the Google team? We’ve been waiting for a resolution on this issue for the past two weeks. Thanks in advance

I’ve already managed to fix that part by forcing the system not to access GCS and instead use the manually downloaded pretrained weights that someone shared with me. So the training now runs without the download error.
However, I’m currently facing another issue: during detection, the generated model is not detecting any objects at all, even though I’m using the same dataset that worked in my previous trainings. So it seems like something might be wrong in one of the training or export steps after bypassing GCS.
If you’ve encountered this before or have any idea what could be causing the detection to fail, please let me know.

**Comment by leonardosarti-nando** [comment]
@sustinespacne9-ai Can you share the weights you are using for the MobilenetV2 SSD model, where did you get them from? I was trying to force the offline model too but couldn't find the correct weights to load

**Comment by sustinespacne9-ai** [comment]
> [@sustinespacne9-ai](https://github.com/sustinespacne9-ai) Can you share the weights you are using for the MobilenetV2 SSD model, where did you get them from? I was trying to force the offline model too but couldn't find the correct weights to load

You can check it here, and just email me if you encounter any problems. https://colab.research.google.com/drive/1JSI29BQZzsME99Ot7gqVkHw5d2EU3wMs?usp=sharing

**Comment by leonardosarti-nando** [comment]
Thanks @sustinespacne9-ai unfortunately for me it's failing during the model._create_model() call if I try to use those weights.
Anyways @HarishPermal @akashvverma1995 any updates on this issue?

**Comment by sustinespacne9-ai** [comment]
> Thanks [@sustinespacne9-ai](https://github.com/sustinespacne9-ai) unfortunately for me it's failing during the model._create_model() call if I try to use those weights. Anyways [@HarishPermal](https://github.com/HarishPermal) [@akashvverma1995](https://github.com/akashvverma1995) any updates on this issue?

https://colab.research.google.com/drive/1WxdGtORqeJSQGaqsH_8D1TIWEqlSA3aG?usp=sharing

Try this instead, this working fine on my side. The detection pipeline runs properly, and the model is able to produce outputs without runtime issues.
However, since I trained this using a small dataset for testing purposes, the accuracy is not yet optimal. To significantly improve detection performance and reduce false positives or random detections, we will need a much larger and more diverse dataset for training. Increasing the number of images per class, improving annotation quality, and including more variations (lighting, angles, backgrounds, object sizes, etc.) should greatly enhance the model’s robustness.
At this stage, the framework and deployment workflow are functioning correctly. The next major step is scaling the dataset and retraining to achieve higher accuracy and more stable predictions.
Feel free to email me if you have any further concerns.

**Comment by schmidt-sebastian** [comment]
The file should be accessible again.
